buildSrc/src/main/java/com/google/firebase/gradle/plugins/ReleaseGenerator.kt

```diff
@@ -31,25 +31,69 @@ import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 import org.gradle.kotlin.dsl.findByType
 
+data class ReleaseReport(
+  val changes: Map<String, List<CommitDiff>>,
+  val changedLibrariesWithNoChangelog: Set<String>
+) {
+  override fun toString() =
+    """
+      |# Release Report
+      |${
+    changes.entries.joinToString("\n") {
+      """
+      |## ${it.key}
+      
+      |${it.value.joinToString("\n") { it.toString() }}
+      """.trimMargin()
+    }
+  }
+      |
+      |## SDKs with changes, but no changelogs
+      |${changedLibrariesWithNoChangelog.joinToString("  \n")}
+    """
+      .trimMargin()
+}
+
 data class CommitDiff(
   val commitId: String,
+  val prId: String,
   val author: String,
   val message: String,
+  val commitLink: String,
+  val prLink: String,
 ) {
-  constructor(
-    revCommit: RevCommit
-  ) : this(revCommit.id.name, revCommit.authorIdent.name, revCommit.fullMessage) {}
+  companion object {
+    private val PR_ID_EXTRACTOR = Regex(".*\\(#(\\d+)\\).*")
```

Comment:
Could you provide documentation with an example of valid input? It doesn't need to be [as extraneous as this](https://github.com/firebase/firebase-android-sdk/blob/7b940f822555f554318cd7df64d6a67537bdc3e2/buildSrc/src/main/java/com/google/firebase/gradle/plugins/Changelog.kt#L190)- but enough to get an idea of what it's doing.